src/senaite/core/upgrade/v02_05_000.py

```diff
@@ -635,3 +626,27 @@ def fix_range_values_for(brains):
         if reindex:
             obj.reindexObject()
         obj._p_deactivate()
+
+
+def purge_orphan_worksheets(tool):
+    """Walks through all records from worksheets catalog and remove orphans
+    """
+    logger.info("Purging orphan Worksheet records from catalog ...")
+    request = api.get_request()
+    cat = api.get_tool(WORKSHEET_CATALOG)
+    paths = cat._catalog.uids.keys()
+    for path in paths:
+        # try to wake-up the object
+        obj = cat.resolve_path(path)
+        if obj is None:
+            obj = cat.resolve_url(path, request)
+
+        if obj is None:
+            # object is missing, remove
+            logger.info("Removing stale record: {}".format(path))
+            cat.uncatalog_object(path)
```

Comment:
should this be `api.uncatalog_object` instead?

Reply:
No, because we don't have an object here, but a path. `api.uncatalog_object` expects an object

Reply:
Ah, you are right, thanks!